thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java

```diff
@@ -0,0 +1,220 @@
+      throws Exception {
+    ArrayList<Long> cloneFunctionIds = new ArrayList<>();
+    try {
+      Map<Long, String> idNameTagMap = parseFunctionIdsAndNameTags(functionIds, nameTags);
+
+      for (Map.Entry<Long, String> entry : idNameTagMap.entrySet()) {
+        cloneFunctionIds.add(cloneAnomalyFunctionById(entry.getKey(), entry.getValue(), Boolean.valueOf(cloneAnomaly)));
+      }
+    } catch (Exception e) {
+      LOG.error("Can not clone function Ids {}, with name tags {}", functionIds, nameTags);
+      throw new WebApplicationException(e);
+    }
+    return cloneFunctionIds;
+  }
+
+  // clone function 1 by 1
+  @POST
+  @Path("function/{id}/clone/{tag}")
+  public Long cloneFunctionsGetIds(@PathParam("id") Long id,
+      @PathParam("tag") String tag,
+      @QueryParam("cloneAnomaly")@DefaultValue("false") String cloneAnomaly)
+      throws Exception {
+    try {
+      return cloneAnomalyFunctionById(id, tag, Boolean.valueOf(cloneAnomaly));
+    } catch (Exception e) {
+      LOG.error("Can not clone function: Id {}, with name tag {}", id, tag);
+      throw new WebApplicationException(e);
+    }
+  }
+
+
+  // util functions for clone anomaly functions
+
+  /**
+   * Parse function ids to be cloned and clone name tags together
+   * the length should be aligned otherwise name tags are all empty
+   * or all replace with the first name tag (if length == 1)
+   *
+   * @param functionIds ',' separated string with numbers representing function ids to be cloned
+   * @param nameTags ',' separated string with strings representing name tags for clone functions
+   * @return a HashMap from function id to be cloned and the new name tag
+   */
+  private Map<Long, String> parseFunctionIdsAndNameTags(String functionIds, String nameTags) {
+    List<Long> functionIdsList = new ArrayList<>();
+    if (StringUtils.isNotBlank(functionIds)) {
+      String[] tokens = functionIds.split(",");
+      for (String token : tokens) {
+        functionIdsList.add(Long.valueOf(token));
+      }
+    }
+
+    int len = functionIdsList.size();
+
+    List<String> nameTagList = new ArrayList<>();
+    if (StringUtils.isNotBlank(nameTags)) {
+      String[] tags = nameTags.split(",");
+      if (tags.length == 1) {
+        for (int i = 0; i < len; i++) {
+          nameTagList.add(tags[0]);
+          LOG.debug("only 1 tag, use the tag for all function");
+        }
+      } else {
+        if (tags.length == len) {
+          for (String tag : tags) {
+            nameTagList.add(tag);
+          }
+        } else {
+          LOG.debug("tag list and function id list does not mach, use empty strings as tags");
+          for (int i = 0; i < len; i++) {
+            nameTagList.add("");
+          }
+        }
+      }
+    }
+
+    LOG.info("function ids set: {}", functionIds);
+    LOG.info("name tag set: {}", nameTagList);
+    // Construct Return results
+    HashMap<Long, String> IdNameTagMap = new HashMap<>();
```

Comment:
sorry for not address this comments before merge, will do it next time when working on similar changes.